packages/app/components/stackbox/Stackbox.tsx

```diff
@@ -51,39 +49,68 @@ const balanceOptions = [
 
 export const Stackbox = () => {
   const searchTokenBarRef = useRef<HTMLInputElement>(null);
-  const [isPickingTokenFrom, setIsPickingTokenFrom] = useState<boolean>(false);
-  const [tokenFrom, setTokenFrom] = useState<TokenFromTokenlist>();
-  const [tokenTo, setTokenTo] = useState<TokenFromTokenlist>();
+  const [isPickingFromToken, setIsPickingFromToken] = useState<boolean>(false);
+  const [fromToken, setFromToken] = useState<TokenFromTokenlist>();
+  const [toToken, setToToken] = useState<TokenFromTokenlist>();
   const { closeModal, openModalId, openModal } = useModalContext();
 
   const { chain } = useNetwork();
   const { address } = useAccount();
   const { data: balance } = useBalance({
-    address: Boolean(tokenFrom) ? address : undefined,
-    token: tokenFrom?.address as `0x${string}`,
+    address: Boolean(fromToken) ? address : undefined,
+    token: fromToken?.address as `0x${string}`,
     chainId: chain?.id,
   });
   const [tokenAmount, setTokenAmount] = useState("");
 
-  const [frequency, setFrequency] = useState<string>(HOUR_OPTION);
+  const [frequency, setFrequency] = useState<FREQUENCY_OPTIONS>(
+    FREQUENCY_OPTIONS.hour
+  );
   const [startDateTime, setStartDateTime] = useState<Date>(
     new Date(startDateTimeTimestamp)
   );
   const [endDateTime, setEndDateTime] = useState<Date>(
     new Date(endDateByFrequency[frequency])
   );
 
-  const openConfirmStack = () => openModal(ModalId.CONFIRM_STACK);
-  const openTokenPicker = (isTokenFrom = true) => {
-    setIsPickingTokenFrom(isTokenFrom);
-    openModal(ModalId.TOKEN_PICKER);
-  };
-  const selectToken = isPickingTokenFrom ? setTokenFrom : setTokenTo;
+  const [showTokenAmountError, setShowTokenAmountError] = useState(false);
+  const [showDateTimeError, setDateTimeError] = useState(false);
 
   useEffect(() => {
     setEndDateTime(new Date(endDateByFrequency[frequency]));
   }, [frequency]);
 
+  const openTokenPicker = (isFromToken = true) => {
+    setIsPickingFromToken(isFromToken);
+    openModal(ModalId.TOKEN_PICKER);
+  };
+
+  const openConfirmStack = () => {
+    const endTimeBeforeStartTime =
+      endDateTime.getTime() <= startDateTime.getTime();
+    const tokenAmountIsZero = tokenAmount === "0";
+
+    if (!tokenAmount) {
+      setShowTokenAmountError(true);
+    }
+
+    if (endTimeBeforeStartTime) {
+      setDateTimeError(true);
+    }
+
+    if (
+      fromToken &&
+      toToken &&
+      !endTimeBeforeStartTime &&
```

Comment:
_just an `OCD` thing (feel free to ignore)_

Can we move this condition `!endTimeBeforeStartTime`  to be the last thing?

Find it easier to read.

Reply:
Fixed in c9a830dae34db6f75e49a63d9b2c0a972db77ef6